server/src/test/java/com/objectcomputing/checkins/services/skills/SkillControllerTest.java

```diff
@@ -66,152 +56,180 @@ public void testGETNonExistingEndpointReturns404() {
+
+        assertEquals(Set.of(skill), response.body());
+        assertEquals(HttpStatus.OK,response.getStatus());
 
-        final HttpResponse<?> response = client.toBlocking()
-                .exchange(HttpRequest
-                        .GET(String.format("/?name=%s", testSkillName)));
-        assertEquals(HttpStatus.OK, response.getStatus());
-        response.equals(skill);
     }
 
     @Test
-    public void testGETFindByValue_Pending() {
+    public void testGETFindByValuePending() {
 
-        boolean testPending = false;
-        Skill skill = new Skill();
-        skill.setPending(testPending);
-        List<Skill> result = new ArrayList<Skill>();
-        result.add(skill);
+        Skill skill = createADefaultSkill();
+        final HttpRequest<Object> request = HttpRequest.
+                GET(String.format("/?pending=%s", encodeValue(String.valueOf(skill.isPending())))).basicAuth(MEMBER_ROLE,MEMBER_ROLE);
 
-        when(mockSkillServices.findByValue("testSkill", testPending)).thenReturn(result);
+        final HttpResponse<Set<Skill>> response = client.toBlocking().exchange(request, Argument.setOf(Skill.class));
 
-        final HttpResponse<?> response = client.toBlocking()
-                .exchange(HttpRequest
-                        .GET(String.format("/?pending=%b", testPending)));
-        assertEquals(HttpStatus.OK, response.getStatus());
-        response.equals(skill);
+        assertEquals(Set.of(skill), response.body());
+        assertEquals(HttpStatus.OK,response.getStatus());
 
     }
 
     @Test
-    public void testGETGetById_HappyPath() {
+    public void testGETGetByIdHappyPath() {
+
+        Skill skill = createADefaultSkill();
+
+        final HttpRequest<Object> request = HttpRequest.
+                GET(String.format("/%s", skill.getSkillid())).basicAuth(MEMBER_ROLE,MEMBER_ROLE);
+
+        final HttpResponse<Skill> response = client.toBlocking().exchange(request, Skill.class);
+
+        assertEquals(skill, response.body());
+        assertEquals(HttpStatus.OK,response.getStatus());
 
-        UUID uuid = UUID.fromString(fakeUuid);
-        Skill skill = new Skill();
-        skill.setSkillid(uuid);
-        skill.setName(testSkillName);
-        skill.setPending(pending);
-        List<Skill> result = new ArrayList<Skill>();
-        result.add(skill);
-
-        skill.setSkillid(UUID.fromString(fakeUuid));
-        when(mockSkillServices.readSkill(uuid)).thenReturn(skill);
-
-        final HttpResponse<?> response = client.toBlocking()
-                .exchange(HttpRequest
-                        .GET(String.format("/%s", skill.getSkillid())));
-        assertEquals(HttpStatus.OK, response.getStatus());
-        response.equals(skill);
     }
 
     @Test
     public void testPOSTCreateASkill() {
 
-        Skill testSkill = new Skill("testName2", pending);
-
-        when(mockSkillServices.saveSkill(testSkill)).thenReturn(testSkill);
+        Skill skill = createADefaultSkill();
```

Comment:
This test is to create a skill so we don't need a default skill to be created.

Reply:
done

Reply:
done